linera-core/src/client/mod.rs

```diff
@@ -1723,9 +1711,14 @@ where
                 .await
             {
                 if let Some(blob_ids) = original_err.get_blobs_not_found() {
-                    self.update_local_node_with_blobs_from(blob_ids, remote_node)
-                        .await?;
-                    continue; // We found the missing blobs: retry.
+                    if let Err(error) = self
+                        .update_local_node_with_blobs_from(blob_ids, remote_node)
+                        .await
+                    {
+                        warn!("Error updating local node with blobs: {error}");
```

Comment:
So we don't propagate this error anymore? Why is that safe?

Reply:
Before, we were converting all kinds of errors (including local ones) that would happen in `update_local_node_with_blobs_from` into `NodeErrors`.

I agree this here isn't ideal either. I can look into this more closely tomorrow and see if there's a way to clearly distinguish local vs. remote errors here.

Reply:
I think this will be quite a large PR in itself, so I created https://github.com/linera-io/linera-protocol/issues/2875 for it.